src/layout/Custom/CustomWebComponent.tsx

```diff
@@ -48,15 +49,23 @@ export function CustomWebComponent({
     if (current) {
       const handleChange = (customEvent: CustomEvent) => {
         const { value, field } = customEvent.detail;
-        setValue(field, value);
+        if (!dataModelBindings?.simpleBinding && !field) {
+          throw new Error(
+            'If you are not using simpleBinding, you need to include a field in your change event to indicate what datamodel binding you want to save to. See docs: https://github.com/Altinn/altinn-studio/issues/8681',
+          );
+        }
+        if (field) {
+          setValue(field, value);
+        }
+        setValue('simpleBinding', value);
```

Comment:
This always sets `simpleBinding` anyway
```suggestion
        if (field) {
          setValue(field, value);
        } else {
          setValue('simpleBinding', value);
        }
```

Reply:
Good catch :)